analysis/quantifier-comparisons/08-genes-detected-comparison.Rmd

```diff
@@ -0,0 +1,405 @@
+```
+
+```{r}
+# get unique gene lists by getting genes that overlap with all other tools first and then the set difference with the list of genes for that tool
+all_cellranger_overlap <- rowdata_filtered %>%
+  dplyr::filter(gene_id %in% cellranger_genes & tool =="alevin-fry") %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+cellranger_unique <- setdiff(cellranger_genes, all_cellranger_overlap)
+
+cr_overlap <- rowdata_filtered %>%
+  dplyr::filter(gene_id %in% cr_like_genes & alevin_resolution %in% c("cr-like-em", "not_alevin")) %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+cr_unique <- setdiff(cr_like_genes, cr_overlap)
+
+cr_like_overlap <- rowdata_filtered %>%
+  dplyr::filter(gene_id %in% cr_em_genes & alevin_resolution %in% c("cr-like", "not_alevin")) %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+cr_like_unique <- setdiff(cr_em_genes, cr_like_overlap)
+```
+
+
+```{r}
+# visualize overlap between gene sets based on tool
+rowdata_filtered[which(rowdata_filtered$alevin_resolution == "not_alevin"), "alevin_resolution"] <- "cellranger"
+
+gene_detect_df <- rowdata_filtered %>%
+  group_by(gene_id) %>%
+  dplyr::summarise(tools_detected = list(unique(alevin_resolution)))
+
+gene_detect_df %>%
+  dplyr::distinct(gene_id, .keep_all = TRUE) %>%
+  ggplot(aes(x = tools_detected)) +
+  geom_bar() +
+  scale_x_upset(n_intersections = 6)
+```
+
+
+When looking at shared cells only and genes that are expressed and detected > 0, we see that alevin-fry detects more genes than cellranger in both methods. 
+Almost every gene that is found in cellranger is also being found in Alevin-fry so we are probably not losing much.
+We also see here that there are more in `cr-like-em` than in `cr-like`. 
+
+Perhaps the genes that are different are lowly expressed genes and it wouldn't matter anyway. 
+
+If we now look at the gene expression for those genes that are unique to each tool, how do they compare to the gene expression for those genes that are shared?
+
+
+```{r}
+# calculate mean expression of log transformed counts matrix 
+log_row_means <- function(sce){
+   rowMeans(log1p(counts(sce))) %>%
+    as.data.frame() %>%
+    tibble::rownames_to_column("gene_id") %>%
+    dplyr::rename("mean" = ".")
+}
+
+# get list of row mean data frames 
+mean_exp_list <- sces_filtered %>%
+  purrr::map(
+    ~ purrr::map(.x, log_row_means)
+  )
+
+# join together into larger data frame with quant_dir and alevin_resolution as unique ID's
+mean_exp_df <- mean_exp_list %>%
+  purrr::map_df(
+    ~ purrr::map_df(.x, rbind, .id = "quant_dir"), 
+    .id = "alevin_resolution"
+  ) %>%
+  dplyr::left_join(quant_info)
+
+# join back with quant_info to get sample information 
+```
+
+
+```{r}
+# add a column to classify as shared or unique to that tool
+mean_exp_df <- mean_exp_df %>%
+  dplyr::mutate(gene_status = ifelse(gene_id %in% c(cellranger_cr_intersect, cellranger_cr_em_intersect, cr_cr_em_intersect), "shared", "unique")) 
```

Comment:
When I was looking at this plot, I spend a lot of time thinking about why the cellranger graphs were higher, and I think it is because what I was interpreting "shared" as was shared across all, but that is not what you mean here. The confusion is due to all those genes that are found by alevin but not cellranger, most of which are probably because of annotation list differences. I think cellranger is more conservative it what it includes as a "gene" before even mapping. That could explain a lot of the difference. In any case, I think we can make this more explicit and informative with the following change (though I am not sold on these specific labels, and the code is not tested!):
```suggestion
mean_exp_df <- mean_exp_df %>%
  dplyr::mutate(gene_status = dplyr::case_when(
  gene_id %in% intersect(cellranger_cr_intersect, cellranger_cr_em_intersect) ~ "common",
  gene_id %in% c(cellranger_cr_intersect, cellranger_cr_em_intersect, cr_cr_em_intersect) ~ "shared", 
  TRUE ~ "unique")) 
```